src/Framework/TestCase.php

```diff
@@ -2441,7 +2441,7 @@ private function registerMockObjectsFromTestArguments(array $testArguments, arra
                     }
 
                     $this->registerMockObject($testArgument);
-                } elseif (is_array($testArgument) && !in_array($testArgument, $visited)) {
+                } elseif (is_array($testArgument) && !in_array($testArgument, $visited, true)) {
```

Comment:
nice fix

@sebastianbergmann , would you consider applying strictness for `is_array` globally ?